mqtt/mqtt-broker/src/configuration.rs

```diff
@@ -1,16 +1,59 @@
-use std::convert::From;
+use std::convert::{From, TryFrom};
 use std::ops::Mul;
-use std::path::Path;
+use std::path::{Path, PathBuf};
 use std::str::FromStr;
 use std::time::Duration;
 
+use crate::{transport::TransportBuilder, Error, ErrorKind};
 use config::{Config, ConfigError, File, FileFormat};
+use failure::ResultExt;
 use lazy_static::lazy_static;
+use native_tls::Identity;
 use regex::Regex;
 use serde::{Deserialize, Deserializer};
+use tracing::info;
 
 pub const DEFAULTS: &str = include_str!("../config/default.json");
 
+#[derive(Debug, Clone, Deserialize)]
+#[serde(rename_all = "snake_case")]
+pub enum Transport {
```

Comment:
nit: I was a bit confused b/w this `Transport` and `transport::Transport`. Maybe `TransportConfig`?

Reply:
being in 'configuration.rs', do you think it is needed? all the rest of the config sections are defined without the 'config' postfix.

Reply:
I'm fine with it, was just nitpicling.